### DISCUSSION TITLE
How can I compile the latest Stockfish from its source codes without embedded NNUE?

### DISCUSSION THREAD

**Discussion by hamzainaan** [official-stockfish/Stockfish#3319] [Q&A] [answered]
000

**Comment by scchess** [comment]
Stockfish offers UCI option for you not to use NNUE.

**Reply by hamzainaan** [reply]
You are right but it compiles with NNUE file (21MB). How can I do if I have the ability to compile without an embedded NNUE file?

**Comment by ianfab** [comment, marked as answer]
You need to enable `NNUE_EMBEDDING_OFF`, see https://github.com/official-stockfish/Stockfish/blob/77eeea407c9d10cf6dbca6f4e07efbbe836c4fb0/src/evaluate.cpp#L46-L47 i.e., add `-DNNUE_EMBEDDING_OFF` to the CXXFLAGS.

**Reply by hamzainaan** [reply]
Thank you so much. I'm so grateful.

**Reply by mrdcvlsc** [reply]
> You need to enable `NNUE_EMBEDDING_OFF`, see
> 
> https://github.com/official-stockfish/Stockfish/blob/77eeea407c9d10cf6dbca6f4e07efbbe836c4fb0/src/evaluate.cpp#L46-L47
> 
> i.e., add `-DNNUE_EMBEDDING_OFF` to the CXXFLAGS.

Noob question, how can I add this flag using only make commands?

```shell
make -j4 build ARCH=x86-64 COMP=clang COMPCXX=clang++
```

**Reply by vondele** [reply]
`CXXFLAGS=-DNNUE_EMBEDDING_OFF make -j4 build ARCH=x86-64 COMP=clang COMPCXX=clang++`

**Comment by mrdcvlsc** [comment]
It's been a while but I just want to also share this.

When **using make** to **build stockfish** and if you want to **keep the original values** of the `CXXFLAGS` while disabling the the **NNUE** embedding, you better use (`+=`) `CXXFLAGS+=-DNNUE_EMBEDDING_OFF` instead of (`=`) `CXXFLAGS=-DNNUE_EMBEDDING_OFF` at least in my observation, because the latter seems to overwrite the `CXXFLAGS` values.

So it would be something like this:

```cmd
make -j4 build ARCH=x86-64 COMP=clang COMPCXX=clang++ CXXFLAGS+=-DNNUE_EMBEDDING_OFF
```
